tests/integration/local/invoke/test_invoke_cdk_templates_with_function_id.py

```diff
@@ -0,0 +1,63 @@
+import json
+import shutil
+from pathlib import Path
+
+import docker
+import pytest
+from docker.errors import APIError
+from parameterized import parameterized
+
+from samcli import __version__ as version
+from samcli.local.docker.lambda_image import RAPID_IMAGE_TAG_PREFIX
+from tests.integration.local.invoke.invoke_integ_base import InvokeIntegBase
+from samcli.lib.utils.architecture import X86_64
+
+
+class TestCDKSynthesizedTemplatesFunctionIdentifies(InvokeIntegBase):
+
+    template = Path("cdk/cdk_function_id_template.yaml")
+
+    @classmethod
+    def setUpClass(cls):
+        # Run sam build first to build the image functions
+        # We only need to create these images once
+        # We remove them after they are no longer used
+        super(TestCDKSynthesizedTemplatesFunctionIdentifies, cls).setUpClass()
+        build_command_list = super().get_build_command_list(cls, template_path=cls.template_path)
+        super().run_command(cls, command_list=build_command_list)
+
+    def tearDown(self) -> None:
+        # Tear down a unique image resource after it is finished being used
+        docker_client = docker.from_env()
+        try:
+            to_remove = self.teardown_function_name
+            docker_client.api.remove_image(f"{to_remove.lower()}")
+            docker_client.api.remove_image(f"{to_remove.lower()}:{RAPID_IMAGE_TAG_PREFIX}-{version}-{X86_64}")
+        except (APIError, AttributeError):
+            pass
+
+        try:
+            # We don't actually use the build dir so we don't care if it's removed before another process finishes
+            shutil.rmtree(str(Path().joinpath(".aws-sam")))
+        except FileNotFoundError:
+            pass
+
+    @parameterized.expand([
+        ("StandardZipFunctionWithFunctionName", "ThisIsHelloWorldFunction", "LambdaWithFunctionName"),
+    ])
+    @pytest.mark.flaky(reruns=3)
+    def test_invoke_function_with_function_id(self, logical_id, function_name, function_id):
```

Comment:
Should we have any integs tests around the precedence order or different edge cases we may encounter with templates?

Reply:
I have added integration test cases for invoking functions in the nested stacks.
For the precedence on the env var overrides, the cases are unit tested:
https://github.com/aws/aws-sam-cli/pull/3498/files#diff-09b566488e9a422abe5c417dd6065f26119bd9bca9e9524e379c0534258a539fR223